prototypes/ItemRepeaterPrototypes/ItemsRepeaterExperiments/AttachedBehaviors/SelectionModelPage.xaml.cs

```diff
@@ -0,0 +1,134 @@
+﻿using ItemsRepeaterExperiments.Common;
+using Microsoft.UI.Xaml.Controls;
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO;
+using System.Linq;
+using System.Runtime.InteropServices.WindowsRuntime;
+using Windows.Foundation;
+using Windows.Foundation.Collections;
+using Windows.UI.Xaml;
+using Windows.UI.Xaml.Controls;
+using Windows.UI.Xaml.Controls.Primitives;
+using Windows.UI.Xaml.Data;
+using Windows.UI.Xaml.Input;
+using Windows.UI.Xaml.Media;
+using Windows.UI.Xaml.Navigation;
+
+// The Blank Page item template is documented at https://go.microsoft.com/fwlink/?LinkId=234238
+
+namespace ItemsRepeaterExperiments.AttachedBehaviors
+{
+    /// <summary>
+    /// An empty page that can be used on its own or navigated to within a Frame.
+    /// </summary>
+    public sealed partial class SelectionModelPage : Page
+    {
+        private List<Team> teams = DataSourceCreator<Team>.CreateRandomizedList(10);
+        public SelectionModelPage()
+        {
+            this.InitializeComponent();
+        }
+
+
+        /// <summary>
+        /// Attach properties to virtualized items and update containers
+        /// </summary>
+        /// <param name="sender"></param>
+        /// <param name="args"></param>
+        private void ItemsRepeater_ElementPrepared(Microsoft.UI.Xaml.Controls.ItemsRepeater sender, Microsoft.UI.Xaml.Controls.ItemsRepeaterElementPreparedEventArgs args)
+        {
+            var control = args.Element as FrameworkElement;
+
+
+            SelectionBehavior.SetIsSelected(control,
+                selectionModel.SelectedItems.Contains(teams[TeamPresenter.GetElementIndex(control)]));
+        }
+
+        private void Control_PointerReleased(object sender, PointerRoutedEventArgs e)
```

Comment:
Could this be wrapped in a behavior as well ? something that ties the selection model to the interactions ?

Reply:
The problem is how we find the SelectionModel from inside a behavior. Should we walk up the tree? We would also need the repeater, so should we find that too by walking up the tree until there is an ItemsRepeater?

Reply:
I'm not a fan of walking the tree. We could put these as attached properties on the element that the behavior can lookup and the attached properties are wired up through bindings when the elements are created (not sure if we can do it from within the template)

Reply:
I've added them now as an attached property we set inside the ElementPrepared. Also that  pointer released logic sits now inside the SelectionBehaviors.